packages/babel-traverse/src/path/modification.js

```diff
@@ -112,7 +116,10 @@ export function insertAfter(nodes) {
     this.parentPath.isExpressionStatement() ||
     this.parentPath.isLabeledStatement()
   ) {
-    return this.parentPath.insertAfter(nodes);
+    // `replaceWithMultiple` requeues if there's a replacement for this.node,
+    // but not for an ancestor's. Set `shouldRequeue` to true so that any replacement
+    // for an ancestor node can be enqueued. Fix #5628 and #5023.
+    return this.parentPath._insertAfter(nodes, true);
```

Comment:
These two cases should be exactly equivalent. In `#insertAfter`, we hit [`parentPath.isExpressionStatement`](https://github.com/babel/babel/pull/5743/files#diff-04e63519c0e77a22a1a20ef218acc048R122), then [`this.isStatementOrBlock()`](https://github.com/babel/babel/pull/5743/files#diff-04e63519c0e77a22a1a20ef218acc048R145). In `#insertBefore`, we'd hit [`parentPath.isExpressionStatement`](https://github.com/babel/babel/pull/5743/files#diff-04e63519c0e77a22a1a20ef218acc048R21), then [`this.isStatementOrBlock()`](https://github.com/babel/babel/pull/5743/files#diff-04e63519c0e77a22a1a20ef218acc048R32).